Sprint №10 HomeWork done.

Vozzhd opened the pull request:
Added:
-new dimension resources
-new xml's for tracks
-new icons
-Recycler view, ViewHolder,adapter
-mock function for make playlist
-Glide library
Changed:
-SearchActivity
-activitySearch

Vozzhd committed:
Sprint №10 HomeWork done.

Added:
-new dimension resources
-new xml's for tracks
-new icons
-Recycler view, ViewHolder,adapter
-mock function for make playlist
-Glide library
Changed:
-SearchActivity
-activitySearch

taikonavt commented on app/src/main/java/com/practicum/playlistmaker/TrackViewHolder.kt:
```diff
@@ -0,0 +1,31 @@
+        trackName.text = model.trackName
+        artistName.text = model.artistName
+        trackTime.text = model.trackTime
+        val corner = itemView.resources.getDimension(R.dimen.corner_radius).toInt()
```
это тоже можно было в инициализацию вынести к вьюхам

Vozzhd replied:
Вынес

taikonavt commented on app/src/main/res/layout/track_view.xml:
```diff
@@ -0,0 +1,70 @@
+        android:layout_gravity="center_vertical"
+        android:layout_marginStart="12dp" />
+
+    <TextView
+        android:id="@+id/trackName"
+        style="@style/RegularTextView"
+        android:layout_width="250dp"
+        android:layout_height="20dp"
+        android:layout_gravity="top"
+
+        android:layout_marginStart="64dp"
+        android:layout_marginTop="12dp"
+
+        android:maxLines="1"
+        android:textColor="?attr/colorOnPrimary" />
```
не хватает обрезания длинной строки тремя точками. Нужно добавить параметр android:ellipsize="end"

taikonavt replied:
если добавить в разметку параметр tools:text="Here Comes The Sun Here Comes The Sun Here Comes The Sun ", то в окне preview можно увидеть как будет выглядеть вьюха с текстом без установки приложения. Параметры tools не влияют на приложение только на preview

taikonavt replied:
для ImageView можно посмотреть как будет выглядеть картинка tools:src="@drawable/arrow_back"

Vozzhd replied:
Исправил, спасибо за советы)

taikonavt commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -54,6 +59,12 @@ class SearchActivity : AppCompatActivity() {
 
+    override fun onResume() {
+        super.onResume()
+        val inputEditText = findViewById<EditText>(R.id.findField)
```
вызов findViewById дорогая операция, лучше не пользоваться ей часто. Чтобы не вызывать ее в onResume можно вынести в глобальное поле private lateinit var inputEditText: EditText, в onCreate оставляем присваивание. Тогда здесь можно будет обращаться к готовой вьюхе. Только нужно помнить про жизненный цикл активити и не обарщаться к вьюхе когда она еще не инициализирована

Vozzhd replied:
Спасибо, вынес, исправил)

taikonavt commented on app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt:
```diff
@@ -54,6 +59,12 @@ class SearchActivity : AppCompatActivity() {
+    override fun onResume() {
+        super.onResume()
+        val inputEditText = findViewById<EditText>(R.id.findField)
+        inputEditText.setSelection(inputEditText.length())
```
не очень понимаю назначение этой строки. Если курсор в inputEditText стоит в середине и мы повернем экран телефона, то курсор переедет в конец

Vozzhd replied:
А если её уберём - то при повороте экрана курсор будет смещаться в начало. Пока нашёл такой способ решения проблемы(

taikonavt commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -2,6 +2,7 @@
 <FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
```
на экране поиска нет стрелки назад в action bar. 
Как вариант action bar можно сделать при помощи toolbar 
https://developer.android.com/reference/android/widget/Toolbar
но и так тоже ок

Vozzhd replied:
Я уже встречал по-моему три варианта дизайна этого экрана)) в одном стрелка есть, в другом - нет. 
Можно конечно каждый раз переделывать, но если можно - я подожду окончательного варианта вёрстки, в проекте в котором собраны все экраны - стрелки нет.

Vozzhd committed:
Sprint №10 HomeWork done after review

Added:
-android:ellipsize="end" for trackName
Changed:
-track_view
-SearchActivity - transform inputEditText to lateinit var for delete findViewById
-TrackViewHolder
